python/katana/local/analytics/_ksssp.pyx

```diff
@@ -0,0 +1,267 @@
+
+    @staticmethod
+    def delta_step_barrier(unsigned delta = kDefaultDelta) -> KssspPlan:
+        """
+        Delta stepping with barrier
+        """
+        return KssspPlan.make(_KssspPlan.DeltaStepBarrier(delta))
+
+
+class _KssspAlgorithmReachability(Enum):
+    """
+    The algorithms to check reachability for kSSSP
+    :see: :py:class: `~katana.local.analytics.AlgoReachability` constructors for algorithm documentation.
+    """
+    AsyncLevel = _AlgoReachability.Algorithm.asyncLevel
+    SyncLevel = _AlgoReachability.Algorithm.syncLevel
+
+
+cdef class AlgoReachability:
+    """
+    The algorithms available to check reachability between two nodes
+    Static method construct AlgoReachability using specific algorithms.
+    """
+
+    cdef:
+        _AlgoReachability underlying_
+
+    cdef _AlgoReachability* underlying(self) except NULL:
+        return &self.underlying_
+
+    @staticmethod
+    cdef AlgoReachability make(_AlgoReachability u):
+        f = <AlgoReachability>AlgoReachability.__new__(AlgoReachability)
+        f.underlying_ = u
+        return f
+
+    def __init__(self):
+        """
+        Construct AlgoReachability using default parameters
+        """
+        self.underlying_ = _AlgoReachability()
+
+    Algorithm = _KssspAlgorithmReachability
+
+    @property
+    def algorithm(self) -> _KssspAlgorithmReachability:
+        """
+        The selected algorithm.
+        """
+        return _KssspAlgorithmReachability(self.underlying_.algorithm())
+
+    def __init__(self):
+        """
+        Choose an algorithm
+        """
+        super(AlgoReachability, self).__init__()
+
+    @staticmethod
+    def async_level() -> AlgoReachability:
+        """
+        Asynchronous level reachability
+        """
+        return AlgoReachability.make(_AlgoReachability.AsyncLevel())
+
+    @staticmethod
+    def sync_level() -> AlgoReachability:
+        """
+        Synchronous level reachability
+        """
+        return AlgoReachability.make(_AlgoReachability.SyncLevel())
+
+
+def ksssp(pg, str edge_weight_property_name, size_t start_node,
+          size_t report_node, size_t num_paths, bool is_symmetric=False,
+          AlgoReachability algo_reachability = AlgoReachability(),
+          KssspPlan plan = KssspPlan().delta_tile(), *, txn_ctx = None):
+    """
+    Compute the K-Shortest Path on `pg` using `start_node` as source.
+
+    :type pg: katana.local.Graph
+    :param pg: The graph to analyze
+    :type edge_weight_property_name: str
+    :param edge_weight_property_name: The input property containing edge weights.
+    :type start_node: Node ID
+    :param start_node: The source node
+    :type report_node: Node ID
+    :param report_node: The destination node
+    :type num_paths: int
+    :param num_paths: Number of paths to look for
+    :type is_symmetric: bool
+    :param is_symmetric: Whether or not the path is symmetric. Defaults to false.
```

Comment:
is_symmetric is for the graph or the path?

Reply:
For the graph, thanks for the catch